op-node/rollup/event/util.go

```diff
@@ -0,0 +1,19 @@
+package event
+
+// Is as helper function is syntax-sugar to do an Event type check as a boolean function
+func Is[T Event](ev Event) bool {
+	_, ok := ev.(T)
+	return ok
+}
```

Comment:
Wouldn't it then just become `rollup.IsResetEvent` instead of `event.Is[rollup.ResetEvent]`? It's shorter, but also need to use it a lot for the extra lines of type definitions to really be worth it.